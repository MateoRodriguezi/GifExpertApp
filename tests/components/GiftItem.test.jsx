import { render, screen } from "@testing-library/react";
import { GiftItem } from "../../src/components/GiftItem";

describe('Pruebas en <GifItem/>', () => { 

    const title = 'Saitama'
    const url = 'https://dragonball.com/goku.jpg'
    
    test ('debe de hacer match contra snapshot', () => {

        const {container} = render(<GiftItem title= {title} url= {url} />)
        expect (container).toMatchSnapshot();

    });

    test ('debe mostrar la imagen con el URL y el ALT indicado', () => {

        render(<GiftItem title= {title} url= {url} />)
        expect ( screen.getByRole('img').src ).toBe(url)
        expect ( screen.getByRole('img').alt ).toBe(title)

    });

    test ('debe mostrar el titulo en el componenete', () => {

        render(<GiftItem title= {title} url= {url} />)
        expect (screen.getByText(title)).toBeTruthy

    });

 });