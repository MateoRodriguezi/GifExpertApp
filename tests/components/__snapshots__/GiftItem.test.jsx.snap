// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifItem/> debe de hacer match contra snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Saitama"
      src="https://dragonball.com/goku.jpg"
    />
    <p>
       
      Saitama
       
    </p>
  </div>
</div>
`;
